packages/web/src/pages/sign-up-page/pages/SignUpPage.tsx

```diff
@@ -34,30 +59,151 @@ type SignUpEmailValues = {
   email: string
 }
 
+const FormSchema = z.object({
+  email: z
+    .string({ required_error: messages.invalidEmail })
+    .regex(EMAIL_REGEX, { message: messages.invalidEmail })
+})
+
 export const SignUpPage = (props: SignUpPageProps) => {
   const { onNext } = props
-
-  const handleSubmit = useCallback(
+  const dispatch = useDispatch()
+  const navigate = useNavigateToPage()
+  const [checkEmailInUse, result] = useEmailInUse()
+  const { data: emailExists, status: emailApiStatus, errorMessage } = result
+  const submitHandler = useCallback(
     (values: SignUpEmailValues) => {
       const { email } = values
-      onNext({ stage: 'create-password', params: { email } })
+      checkEmailInUse(email)
+      dispatch(setValueField('email', email))
     },
-    [onNext]
+    [dispatch, checkEmailInUse]
   )
+  const formikForm = useFormik({
+    initialValues,
+    onSubmit: submitHandler,
+    validationSchema: toFormikValidationSchema(FormSchema),
+    validateOnBlur: true,
+    validateOnChange: false
+  })
+  const {
+    values,
+    handleChange,
+    handleBlur,
+    touched,
+    errors,
+    handleSubmit,
+    setErrors
+  } = formikForm
+
+  const isLoading = emailApiStatus === Status.LOADING
+
+  // Check for API status changes (occurs after form submit)
+  useEffect(() => {
+    // Unknown error state
+    if (emailApiStatus === Status.ERROR) {
+      setErrors({ email: messages.unknownError })
+      return
+    }
+    if (emailApiStatus === Status.SUCCESS) {
+      if (emailExists) {
+        // Redirect to sign in if the email exists already
+        navigate(SIGN_IN_PAGE)
+      } else {
+        // Move onto the password page
+        onNext({ stage: 'create-password' })
+      }
+    }
```

Comment:
Yeah from what I can tell we want to update the value field before:
- moving on to password page
- swapping over to sign in page

I'm actually not sure about moving to sign in page if we want to have it forward along a partial email if the user clicks the "sign in" link? Seems like it could be a nice-to-have UX but I just checked and it doesn't work that way currently. Thoughts?